module-2/backend/week_two.md

```diff
@@ -14,32 +17,106 @@ end
+
 3. Assume that in your database, a team has the following attributes: "id", "name", owner_id". How would you find the name of a team with an id of 4? Assuming your class only included the code from question 2, how could you find the owner of the same team?
 
+  
+ (a)  Team.find(4).name
+ (b)  team_owner = Team.find(4).owner_id
+      Owner.find_by_owner_id(owner_id).name
+  
+
 4. Assume that you added a line to your `Team` class as follows:
 
 ```ruby
 class Team << ActiveRecord::Base
   belongs_to :owner
 end
-```
+```  
 
 Now how would you find the owner of the team with an id of 4?
 
+Team.find_by_id(4).owner_id.name
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+Teachers 
+have_many :students 
+
+Students
+belong_to :teachers
+
+Schema: 
+Teachers
+name: "name"
+
+Students
+name: "name"
+teacher_id: id
+
 6. Define foreign key, primary key, and schema.
+
+foreign key - the id for a class of objects that have many x's, for instance, a student will have the foreign key for a teacher 
+primary key - this is the id for a record in its home table (student id in the student record)
+schema - gives us a framework for figuing out what the relationships between our tables should be. 
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+  The primay key on the teachers table will become the foreign key on the Students table. 
+  
 8. What are the parts of an HTTP response?
 
+  status line, status code, and reason phrase(?) 
+
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+  pluck - selects for certain attribute, like a specific date within the date column. 
+  joins - creates a relationship between two tables, without having the relationships in the schema.
+  where - allows you to specify some condition, and where that condition is met, it will return the record.  
+  
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
+  rake routes - shows your HTTP verb, route, and the view file
+  db:reset - because sometimes your database gets messed up and you have no idea why. 
+  rake --tasks  - because I'm not sure what my favorites are yet :) 
+
 3. What two columns does `t.timestamps null: false` create in our database?
+
+  created_at and updated_at
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+Teachers belong_to schools
+Schools have_many teachers
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+  a column in the teachers' table for the school foreign key?
+  
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+  I'm not sure what this question is asking. 
```

Comment:
Joins tables don't just have to hold the foreign keys of the tables they join. `invoice_items` from Black Thursday is a good example: it holds invoice_id, item_id, quantity, and unit_price.